storage/uploader.go

```diff
@@ -0,0 +1,353 @@
+package storage
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	authhttp "github.com/bsv-blockchain/go-sdk/auth/clients/authhttp"
+)
+
+// StorageUploader implements the StorageUploaderInterface
+type StorageUploader struct {
+	baseURL   string              // Base URL of the storage service
+	authFetch *authhttp.AuthFetch // Authenticated HTTP client for API requests
+}
+
+// NewStorageUploader creates a new uploader instance
+func NewStorageUploader(config UploaderConfig) (*StorageUploader, error) {
+	if config.StorageURL == "" {
+		return nil, errors.New("storage URL is required")
+	}
+	if config.Wallet == nil {
+		return nil, errors.New("wallet is required for authentication")
+	}
+
+	// Create auth fetch client
+	authClient := authhttp.New(config.Wallet, nil, nil)
+
+	return &StorageUploader{
+		baseURL:   config.StorageURL,
+		authFetch: authClient,
+	}, nil
+}
+
+// uploadInfo is used to parse the response from the /upload endpoint
+type uploadInfo struct {
+	Status          string            `json:"status"`
+	UploadURL       string            `json:"uploadURL"`
+	RequiredHeaders map[string]string `json:"requiredHeaders"`
+	Amount          *int64            `json:"amount,omitempty"`
+}
+
+// getUploadInfo requests information from the server to upload a file
+func (u *StorageUploader) getUploadInfo(ctx context.Context, fileSize int, retentionPeriod int) (*uploadInfo, error) {
+	url := fmt.Sprintf("%s/upload", u.baseURL)
+	requestBody := map[string]interface{}{
+		"fileSize":        fileSize,
+		"retentionPeriod": retentionPeriod,
+	}
+
+	requestJSON, err := json.Marshal(requestBody)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal request body: %w", err)
+	}
+
+	reqOpts := &authhttp.SimplifiedFetchRequestOptions{
+		Method: "POST",
+		Headers: map[string]string{
+			"Content-Type": "application/json",
+		},
+		Body: requestJSON,
+	}
+
+	resp, err := u.authFetch.Fetch(ctx, url, reqOpts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get upload info: %w", err)
+	}
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("upload info request failed: HTTP %d", resp.StatusCode)
+	}
+
+	var info uploadInfo
+	if err := json.NewDecoder(resp.Body).Decode(&info); err != nil {
+		return nil, fmt.Errorf("failed to decode upload info response: %w", err)
+	}
+
+	if info.Status == "error" {
```

Comment:
Maybe instead of `== "error"` use `!= "ok"` or whatever the desired value is, so the default is to error. Also, constants for possible values for `info.Status` could be helpful.

This comment applies to multiple locations in `uploader.go`.